LoopKit/Service.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Service.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 5/17/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+
+public protocol ServiceDelegate: AnyObject {
+
+    /// Informs the delegate that the specified service was updated.
+    ///
+    /// - Parameters:
+    ///     - service: The service updated.
+    func serviceUpdated(_ service: Service)
+
+    /// Informs the delegate that the specified service was deleted.
+    ///
+    /// - Parameters:
+    ///     - service: The service deleted.
+    func serviceDeleted(_ service: Service)
+
+}
+
+
+public protocol Service: DeviceManager {
+
+    var serviceDelegate: ServiceDelegate? { get set }
+
+    /// Is the service complete?
+    var isComplete: Bool { get }
```

Comment:
Does "complete" mean properly configured, or?

Reply:
Renamed to `hasValidConfiguration` to be more descriptive.